File: Practica 5/EcuacionCuadraticaPOO.java
import java.util.Scanner;

class EcuacionCuadratica {
    private double a, b, c;

    public EcuacionCuadratica(double a, double b, double c) {
        this.a = a;
        this.b = b;
        this.c = c;
    }

    public double getDiscriminante() {
        return b * b - 4 * a * c;
    }

    public Double getRaiz1() {
        double d = getDiscriminante();
        return d >= 0 ? (-b + Math.sqrt(d)) / (2 * a) : null;
    }

    public Double getRaiz2() {
        double d = getDiscriminante();
        return d >= 0 ? (-b - Math.sqrt(d)) / (2 * a) : null;
    }

    public void resolver() {
        double d = getDiscriminante();
        if (d > 0) {
            System.out.printf("La ecuación tiene dos raíces %.5f y %.5f%n", getRaiz1(), getRaiz2());
        } else if (d == 0) {
            System.out.printf("La ecuación tiene una raíz %.5f%n", -b / (2 * a));
        } else {
            System.out.println("La ecuación no tiene raíces reales");
        }
    }
}

public class EcuacionCuadraticaPOO {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.print("Ingrese a, b, c: ");
        double a = sc.nextDouble();
        double b = sc.nextDouble();
        double c = sc.nextDouble();
        sc.close();

        EcuacionCuadratica ecuacion = new EcuacionCuadratica(a, b, c);
        ecuacion.resolver();
    }
}

File: Practica 5/ecuacion_cuadratica_poo.py
import math

class EcuacionCuadratica:
    def __init__(self, a, b, c):
        self.a = a
        self.b = b
        self.c = c

    def get_discriminante(self):
        return self.b**2 - 4*self.a*self.c

    def get_raiz1(self):
        d = self.get_discriminante()
        return (-self.b + math.sqrt(d)) / (2 * self.a) if d >= 0 else None

    def get_raiz2(self):
        d = self.get_discriminante()
        return (-self.b - math.sqrt(d)) / (2 * self.a) if d >= 0 else None

    def resolver(self):
        d = self.get_discriminante()
        if d > 0:
            print(f"La ecuación tiene dos raíces {self.get_raiz1():.5f} y {self.get_raiz2():.5f}")
        elif d == 0:
            print(f"La ecuación tiene una raíz {-self.b / (2 * self.a):.5f}")
        else:
            print("La ecuación no tiene raíces reales")

a, b, c = map(float, input("Ingrese a, b, c: ").split())
ecuacion = EcuacionCuadratica(a, b, c)
ecuacion.resolver()
